src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -679,6 +681,11 @@ public Topic getDeploymentPollingFrequencySeconds() {
         return getTopic(DEPLOYMENT_POLLING_FREQUENCY_SECONDS);
     }
 
+    public Topic gets3EndpointType() {
+        return getTopic(S3_ENDPOINT_TYPE).dflt(S3_ENDPOINT_DEFAULT_TYPE);
```

Comment:
If a customer did not set nucleus configuration but set the environment variable `aws.s3UseUsEast1RegionalEndpoint`, should we use regional or global s3 endpoint?

From my opinion, if one of the two is set to REGIONAL, then we should use regional endpoint for both gg and s3 apis.

Reply:
Updated here